### ISSUE TITLE
[Bug] IndexError: Dimension out of range (expected to be in range of [-1, 0], but got -2)

### ISSUE THREAD

**Issue by fortunewalla** [pytorch/botorch#315]
# 🐛 Bug
After following conda installation instructions, tried to run in jupyter notebook the example "2. Fit a model:" from the botorch.org frontpage. Got the following error.

## To reproduce

** Code snippet to reproduce **
```python
import torch
from botorch.models import SingleTaskGP
from botorch.fit import fit_gpytorch_model
from gpytorch.mlls import ExactMarginalLogLikelihood

train_X = torch.rand(10, 2)
Y = 1 - torch.norm(train_X - 0.5, dim=-1) + 0.1 * torch.rand(10)
train_Y = (Y - Y.mean()) / Y.std()

gp = SingleTaskGP(train_X, train_Y)
mll = ExactMarginalLogLikelihood(gp.likelihood, gp)
fit_gpytorch_model(mll)
```

** Stack trace/error message **
```bash
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
<ipython-input-2-fd0caaa8f8fc> in <module>
      8 train_Y = (Y - Y.mean()) / Y.std()
      9 
---> 10 gp = SingleTaskGP(train_X, train_Y)
     11 mll = ExactMarginalLogLikelihood(gp.likelihood, gp)
     12 fit_gpytorch_model(mll)

~/miniconda3/envs/j/lib/python3.7/site-packages/botorch/models/gp_regression.py in __init__(self, train_X, train_Y, likelihood, covar_module)
     77             >>> model = SingleTaskGP(train_X, train_Y)
     78         """
---> 79         validate_input_scaling(train_X=train_X, train_Y=train_Y)
     80         self._validate_tensor_args(X=train_X, Y=train_Y)
     81         self._set_dimensions(train_X=train_X, train_Y=train_Y)

~/miniconda3/envs/j/lib/python3.7/site-packages/botorch/models/utils.py in validate_input_scaling(train_X, train_Y, train_Yvar, raise_on_fail)
    219             raise InputDataError("Input data contains negative variances.")
    220     check_min_max_scaling(X=train_X, raise_on_fail=raise_on_fail)
--> 221     check_standardization(Y=train_Y, raise_on_fail=raise_on_fail)

~/miniconda3/envs/j/lib/python3.7/site-packages/botorch/models/utils.py in check_standardization(Y, atol_mean, atol_std, raise_on_fail)
    172     """
    173     with torch.no_grad():
--> 174         Ymean, Ystd = torch.mean(Y, dim=-2), torch.std(Y, dim=-2)
    175         if torch.abs(Ymean).max() > atol_mean or torch.abs(Ystd - 1).max() > atol_std:
    176             msg = (

IndexError: Dimension out of range (expected to be in range of [-1, 0], but got -2)
```

## Expected Behavior
The example should run

## System information

**Please complete the following information:**
- BoTorch Version (`0.1.4`)
- GPyTorch Version (`0.3.6`)
- PyTorch Version (`1.3.1`)
- centos 8
- Linux localhost.localdomain `4.18.0-80.11.2.el8_0.x86_64` #1 SMP Tue Sep 24 11:32:19 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
- conda (`4.7.12`)

## Additional context
```bash
(j) [user@localhost ~]$ conda install botorch -c pytorch -c gpytorch
Collecting package metadata (current_repodata.json): done
Solving environment: done

## Package Plan ##

  environment location: /home/user/miniconda3/envs/j

  added / updated specs:
    - botorch


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    botorch-0.1.4              |                0          96 KB  pytorch
    cffi-1.13.1                |   py37h2e261b9_0         224 KB
    cudatoolkit-10.1.243       |       h6bb024c_0       347.4 MB
    gpytorch-0.3.6             |                0         180 KB  gpytorch
    ninja-1.9.0                |   py37hfd86e86_0         1.2 MB
    pytorch-1.3.1              |py3.7_cuda10.1.243_cudnn7.6.3_0       428.0 MB  pytorch
    ------------------------------------------------------------
                                           Total:       777.1 MB

The following NEW packages will be INSTALLED:

  botorch            pytorch/noarch::botorch-0.1.4-0
  cffi               pkgs/main/linux-64::cffi-1.13.1-py37h2e261b9_0
  cudatoolkit        pkgs/main/linux-64::cudatoolkit-10.1.243-h6bb024c_0
  gpytorch           gpytorch/noarch::gpytorch-0.3.6-0
  ninja              pkgs/main/linux-64::ninja-1.9.0-py37hfd86e86_0
  pycparser          pkgs/main/linux-64::pycparser-2.19-py37_0
  pytorch            pytorch/linux-64::pytorch-1.3.1-py3.7_cuda10.1.243_cudnn7.6.3_0


Proceed ([y]/n)? y


Downloading and Extracting Packages
ninja-1.9.0          | 1.2 MB    | ####################################### | 100%
cffi-1.13.1          | 224 KB    | ####################################### | 100%
botorch-0.1.4        | 96 KB     | ####################################### | 100%
pytorch-1.3.1        | 428.0 MB  | ####################################### | 100%
cudatoolkit-10.1.243 | 347.4 MB  | ####################################### | 100%
gpytorch-0.3.6       | 180 KB    | ####################################### | 100%
Preparing transaction: done
Verifying transaction: done
Executing transaction: done
```

**Comment by Balandat** [comment]
Thanks for reporting this. 

Since 0.1.4 botorch requires explicit output dimensions in the training targets `Y` so as to remove ambiguity when working with batches. We must have missed updating the examples before pushing the last "stable" website. See the "latest" page for the correct form of the example: https://botorch.org/v/latest/index.html

But this report also shows that we're not raising a proper actionable error here b/c we check for standardization/normalization of the input before we check for the dimensions. We'll fix that.

**Comment by fortunewalla** [comment]
Using the code for "2. Fit a model:" from [https://botorch.org/v/latest/index.html](https://botorch.org/v/latest/index.html) it seems to be working. Thanks.

```python
ExactMarginalLogLikelihood(
  (likelihood): GaussianLikelihood(
    (quadrature): GaussHermiteQuadrature1D()
    (noise_covar): HomoskedasticNoise(
      (noise_prior): GammaPrior()
      (raw_noise_constraint): GreaterThan(1.000E-04)
    )
  )
  (model): SingleTaskGP(
    (likelihood): GaussianLikelihood(
      (quadrature): GaussHermiteQuadrature1D()
      (noise_covar): HomoskedasticNoise(
        (noise_prior): GammaPrior()
        (raw_noise_constraint): GreaterThan(1.000E-04)
      )
    )
    (mean_module): ConstantMean()
    (covar_module): ScaleKernel(
      (base_kernel): MaternKernel(
        (lengthscale_prior): GammaPrior()
        (raw_lengthscale_constraint): Positive()
        (distance_module): Distance()
      )
      (outputscale_prior): GammaPrior()
      (raw_outputscale_constraint): Positive()
    )
  )
)
```

Also final output of "4. Optimize the acquisition function:" is
```python
tensor([[0.4753, 0.5200]])
```

**Comment by Balandat** [comment]
Great to hear. We're not fixing the seed here, so this is a completely random problem so the result will be different in each run. 

I'll leave this issue open until we've fixed the error message. Thanks again for reporting.

**Comment by Balandat** [comment]
Closing this as the stable website has been updated and the error messages will be fixed once #368 is merged in.
